### PR TITLE
importccl: EXPORT CSV

### PR DESCRIPTION
**Pull request by dt**
This is a re-opening of #24755 after it was closed by a mistaken branch deletion, which unfortunately also prevents re-opening in place.

This adds a DistSQL processor that writes whatever it is given as CSV files to a specified storage location.

This processor can be appended as a sink for arbitrary SELECT queries.

The primary intended use-case is dumping tables as CSV for bulk-export, but it can also be used on any SELECT query, so it could also be used in report generation or other workflows.

Release note (enterprise change): prototype support for EXPORT CSV.

### CODE REVIEW

**Comment by andreimatei** [comment]
LGTM

**Review by madelynnblue — APPROVED** [review]

File: pkg/ccl/importccl/exportcsv.go [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+		sp.input.Start(ctx)
+		input := distsqlrun.MakeNoMetadataRowSource(sp.input, sp.output)
+
+		alloc := &sqlbase.DatumAlloc{}
+
+		var buf bytes.Buffer
+		writer := csv.NewWriter(&buf)
+		writer.Comma = sp.spec.Delimiter
+
+		f := tree.NewFmtCtxWithBuf(tree.FmtBareStrings)
```
[the remark this answers is not in the document]
Comment (dt) [reply pkg/ccl/importccl/exportcsv.go]: Done

**Comment by mjibson** [comment]
Oh I just remembered about NULLs. We better add a tests for those.

**Comment by dt** [comment]
Done, added tests with both `''` and `NULL` representations for nulls.

**Review by madelynnblue — APPROVED** [review]

**Pull request merged**